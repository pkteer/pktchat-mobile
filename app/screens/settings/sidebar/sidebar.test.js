// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import MainSidebar from '@components/sidebars/main/main_sidebar.ios';
import {DeviceTypes} from '@constants';
import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import SidebarSettings from './index';

jest.mock('app/mattermost_managed', () => ({
    isRunningInSplitView: jest.fn().mockResolvedValue(false),
    addEventListener: jest.fn(),
}));

describe('SidebarSettings', () => {
    const baseProps = {
        theme: Preferences.THEMES.denim,
    };

    test('should match, full snapshot', async () => {
        const wrapper = shallowWithIntl(
            <SidebarSettings {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();

        await wrapper.instance().loadSetting();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should set the Permanent Sidebar value to false', async () => {
        const wrapper = shallowWithIntl(
            <SidebarSettings {...baseProps}/>,
        );

        await wrapper.instance().loadSetting();
        expect(wrapper.state('enabled')).toBe(false);
    });

    test('should set the Permanent Sidebar value to true and update the sidebar', async () => {
        DeviceTypes.IS_TABLET = true;

        const wrapper = shallowWithIntl(
            <SidebarSettings {...baseProps}/>,
        );

        const mainProps = {
            actions: {
                getTeams: jest.fn(),
                makeDirectChannel: jest.fn(),
                setChannelDisplayName: jest.fn(),
                setChannelLoading: jest.fn(),
                joinChannel: jest.fn(),
                setCategoryCollapsed: jest.fn(),
            },
            blurPostTextBox: jest.fn(),
            currentTeamId: 'current-team-id',
            currentUserId: 'current-user-id',
            deviceWidth: 10,
            isLandscape: false,
            teamsCount: 2,
            theme: Preferences.THEMES.denim,
        };

        const mainSidebar = shallowWithIntl(
            <MainSidebar {...mainProps}/>,
        );

        await wrapper.instance().loadSetting();
        expect(wrapper.state('enabled')).toBe(false);

        await wrapper.instance().saveSetting(true);

        expect(wrapper.state('enabled')).toBe(true);
        expect(mainSidebar.state('permanentSidebar')).toBe(true);
    });
});
