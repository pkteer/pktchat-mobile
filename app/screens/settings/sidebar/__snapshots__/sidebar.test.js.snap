// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarSettings should match, full snapshot 1`] = `null`;

exports[`SidebarSettings should match, full snapshot 2`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "backgroundColor": "rgba(63,67,80,0.06)",
        "flex": 1,
        "paddingTop": 35,
      }
    }
  >
    <section
      disableHeader={true}
      theme={
        Object {
          "awayIndicator": "#ffbc1f",
          "buttonBg": "#1c58d9",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3f4350",
          "codeTheme": "github",
          "dndIndicator": "#d24b4e",
          "errorTextColor": "#d24b4e",
          "linkColor": "#386fe5",
          "mentionBg": "#ffffff",
          "mentionColor": "#1e325c",
          "mentionHighlightBg": "#ffd470",
          "mentionHighlightLink": "#1b1d22",
          "newMessageSeparator": "#cc8f00",
          "onlineIndicator": "#3db887",
          "sidebarBg": "#1e325c",
          "sidebarHeaderBg": "#192a4d",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarTeamBarBg": "#14213e",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#5d89ea",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#28427b",
          "sidebarUnreadText": "#ffffff",
          "type": "Denim",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(63,67,80,0.1)",
            "height": 1,
          }
        }
      />
      <sectionItem
        action={[Function]}
        actionType="toggle"
        description={
          <InjectIntl(FormattedText)
            defaultMessage="Keep the sidebar open permanently"
            id="mobile.sidebar_settings.permanent_description"
          />
        }
        label={
          <InjectIntl(FormattedText)
            defaultMessage="Permanent Sidebar"
            id="mobile.sidebar_settings.permanent"
          />
        }
        selected={false}
        theme={
          Object {
            "awayIndicator": "#ffbc1f",
            "buttonBg": "#1c58d9",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3f4350",
            "codeTheme": "github",
            "dndIndicator": "#d24b4e",
            "errorTextColor": "#d24b4e",
            "linkColor": "#386fe5",
            "mentionBg": "#ffffff",
            "mentionColor": "#1e325c",
            "mentionHighlightBg": "#ffd470",
            "mentionHighlightLink": "#1b1d22",
            "newMessageSeparator": "#cc8f00",
            "onlineIndicator": "#3db887",
            "sidebarBg": "#1e325c",
            "sidebarHeaderBg": "#192a4d",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarTeamBarBg": "#14213e",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#5d89ea",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#28427b",
            "sidebarUnreadText": "#ffffff",
            "type": "Denim",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(63,67,80,0.1)",
            "height": 1,
          }
        }
      />
    </section>
  </View>
</RNCSafeAreaView>
`;
